base/compiler/ssair/ir.jl

```diff
@@ -514,21 +514,18 @@ end
     elseif isa(stmt, GotoIfNot)
         op == 1 || throw(BoundsError())
         stmt = GotoIfNot(v, stmt.dest)
-    elseif isa(stmt, EnterNode)
-        op == 1 || throw(BoundsError())
-        stmt = EnterNode(stmt.catch_dest, v)
```

Comment:
This should be `stmt.scope` now, right (like `_useref_getindex`)?

Reply:
It's not merged yet (https://github.com/JuliaLang/julia/pull/52309). But we probably should keep this handling.

Reply:
I think https://github.com/JuliaLang/julia/pull/52309 is good to go, so let me just merge that and then this can be rebased and reviewed on top.

Reply:
Ok, so I guess no changes are required on this part now. This PR does some minor followups on #52309.